h2o-core/src/test/java/water/persist/PersistHTTPTest.java

```diff
@@ -0,0 +1,99 @@
+package water.persist;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.*;
+import water.fvec.*;
+import water.parser.ParseDataset;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+
+import static org.junit.Assert.*;
+
+public class PersistHTTPTest extends TestUtil {
+
+  @BeforeClass
+  public static void setup() {
+    stall_till_cloudsize(1);
+  }
+
+  @Test
+  public void importFilesEager() throws Exception {
+    try {
+      Scope.enter();
+
+      Frame f = Scope.track(parse_test_file(Key.make("prostate.hex"), "smalldata/prostate/prostate.csv"));
+
+      final String localUrl = H2O.getURL(H2O.ARGS.jks != null ? "https" : "http") + "/3/DownloadDataset.bin?frame_id=" + f._key.toString();
+
+      PersistHTTP p = new PersistHTTP();
+
+      assertEquals(-1L, p.checkRangeSupport(URI.create(localUrl))); // H2O doesn't support byte-ranges
+
+      ArrayList<String> files = new ArrayList<>();
+      ArrayList<String> keys = new ArrayList<>();
+      ArrayList<String> fails = new ArrayList<>();
+      ArrayList<String> dels = new ArrayList<>();
+      p.importFiles(localUrl, null, files, keys, fails, dels);
+
+      assertTrue(fails.isEmpty());
+      assertTrue(dels.isEmpty());
+      assertEquals(Collections.singletonList(localUrl), files);
+      assertEquals(Collections.singletonList(localUrl), keys);
+
+      Key<Frame> k = Key.make(localUrl);
+      Frame imported = k.get();
+      assertEquals(1, imported.numCols());
+      assertTrue(imported.vec(0) instanceof UploadFileVec); // Dataset was uploaded eagerly
+
+      Key<Frame> out = Key.make();
+      Frame parsed = Scope.track(ParseDataset.parse(out, k));
+
+      assertTrue(isBitIdentical(f, parsed));
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test
+  public void importFilesLazy() throws Exception {
+    try {
+      Scope.enter();
+
+      Frame f = Scope.track(parse_test_file(Key.make("prostate.hex"), "smalldata/prostate/prostate.csv"));
+
+      final String remoteUrl = "https://h2o-public-test-data.s3.amazonaws.com/smalldata/prostate/prostate.csv";
+      final long expectedSize = 9254L;
+
+      PersistHTTP p = new PersistHTTP();
+
+      assertEquals(expectedSize, p.checkRangeSupport(URI.create(remoteUrl))); // S3 supports byte-ranges
+
+      ArrayList<String> files = new ArrayList<>();
+      ArrayList<String> keys = new ArrayList<>();
+      ArrayList<String> fails = new ArrayList<>();
+      ArrayList<String> dels = new ArrayList<>();
+      p.importFiles(remoteUrl, null, files, keys, fails, dels);
+
+
+      Key<Frame> k = Key.make(remoteUrl);
+      Frame imported = Scope.track(k.get());
+      assertEquals(1, imported.numCols());
+      assertTrue(imported.vec(0) instanceof HTTPFileVec);
+
+      ((HTTPFileVec) imported.vec(0)).setChunkSize(imported, 1024);
```

Comment:
It is just for the purposes of this test. H2O handles this automatically based on the size of the data and number of available cores.

I want to make sure I will have more than a single chunk and I don't want to download too much data from the internet. That is why I set the size explicitly - it will give me a predictable chunk layout regardless of where the code is running.

Reply:
Perfect, thanks for great explanation